### PR TITLE
Add ClusterRole For dedicated-admins for CustomDomain CR

### PR DESCRIPTION
**Pull request by dustman9000**
This permits dedicated-admins to create/update/delete/etc. the CustomDomain CR.

### CODE REVIEW

**Review by jharrington22 — APPROVED** [review]
/lgtm

**Comment by jharrington22** [comment]
@jewzaam can I get you to take a quick look

File: deploy/osd-custom-domains/01-dedicated-admins-cluster.ClusterRole.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: osd-custom-domains-dedicated-admin-cluster
```
Comment (jewzaam) [inline deploy/osd-custom-domains/01-dedicated-admins-cluster.ClusterRole.yaml:4]:
```suggestion
  labels:
    managed.openshift.io/aggregate-to-dedicated-admins: "cluster"
  name: osd-custom-domains-dedicated-admin-cluster
```

And remove the ClusterRoleBinding.  The label will aggregate to the `dedicated-admins-cluster` ClusterRole and it'll Just Work.

**Review by jewzaam — CHANGES_REQUESTED** [review]

**Pull request merged**